### PR TITLE
Support mmocr with NPU backend

### PR DESCRIPTION
**Pull request by LRJKD**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Added ascending device support in mmocr.

## Modification

The main modification points are as follows:
We added an NPU device in the DDP scenario and DP scenario when using the NPU.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.
None

## Use cases (Optional)

We tested crnn_academic_dataset.py.

## Checklist

1.Pre-commit or other linting tools are used to fix the potential lint issues.
2.The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3.If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4.The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -4,8 +4,8 @@ repos:
     rev: 5.0.4
     hooks:
       - id: flake8
-  - repo: https://github.com/PyCQA/isort
-    rev: 5.10.1
+  - repo: https://gitee.com/openmmlab/mirrors-isort
```
Comment (gaotongxiao) [inline .pre-commit-config.yaml:7]:
```suggestion
  - repo: https://github.com/PyCQA/isort
```

File: mmocr/utils/util_distribution.py
Comment (gaotongxiao) [inline mmocr/utils/util_distribution.py:1]: Rename to dist_util.py

File: mmocr/apis/train.py [hunk 2]
```diff
@@ -15,7 +16,7 @@
 from mmocr import digit_version
 from mmocr.apis.utils import (disable_text_recog_aug_test,
                               replace_image_to_tensor)
-from mmocr.utils import get_root_logger
+from mmdet.utils import (build_ddp, build_dp, get_root_logger)
```
Comment (gaotongxiao) [inline mmocr/apis/train.py:19]:
```suggestion
from mmocr.utils import (build_ddp, build_dp, get_root_logger)
```

**Review by gaotongxiao — APPROVED** [review]

**Comment by OpenMMLab-Assistant005** [comment]
Hi @LRJKD ！We are grateful for your efforts in helping improve this open-source project during your personal time. 
Welcome to join OpenMMLab Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA
If you have a WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**